src/index.ts

```diff
@@ -1,87 +1,70 @@
-import {API, APIClient, APIClientOptions, FetchProvider, UInt128} from '@wharfkit/antelope'
-import BN from 'bn.js'
+import {APIClient, Asset, AssetType, Name, NameType} from '@wharfkit/antelope'
+import * as SystemTokenContract from './contracts/system.token'
+import {Contract} from '@wharfkit/contract'
 
-import {PowerUpAPI} from './powerup'
-import {RAMAPI} from './ram'
-import {REXAPI} from './rex'
-
-export * from './powerup'
-export * from './ram'
-export * from './rex'
-
-interface ResourcesOptions extends APIClientOptions {
-    api?: APIClient
-    sampleAccount?: string
-    symbol?: string
-    url?: string
-}
-
-export interface SampleUsage {
-    account: API.v1.AccountObject
-    cpu: UInt128
-    net: UInt128
+interface TokenOptions {
+    client: APIClient
+    tokenSymbol?: Asset.SymbolType
+    contract?: Contract
 }
 
-export const BNPrecision = new BN(1000 * 1000)
+export class Token {
+    readonly contract: Contract
+    readonly tokenSymbol: Asset.SymbolType
 
-export class Resources {
-    static __className = 'Resources'
-
-    readonly api: APIClient
-
-    // the account to use when sampling usage
-    sampleAccount = 'b1'
+    constructor({contract, client, tokenSymbol}: TokenOptions) {
+        this.contract = contract || new SystemTokenContract.Contract({client})
+        this.tokenSymbol = tokenSymbol || '4,EOS'
```

Comment:
I am not sure I like that the token symbol defaults to EOS. That's caused me some headaches in the past with our other libraries that "default" to something. 

With that said, the logical next step would be requiring the `tokenSymbol` in the constructor, which might not be great either. 

Looking at the code though - the only reason we have a `tokenSymbol` value is so people can pass numbers in to the `transfer` call, which itself may have some problems (e.g. developer passes in `1.2345` and it turns into `1.234`, unexpected!). So probably not wanting that... I'd say we require an `AssetType` for the `amount` value on the `transfer` action, and we don't need the `tokenSymbol` anymore.

Reply:
I did some quick research and my understanding is that 4 decimal numbers can be represented quite accurately as a number in Javascript. I think you have a good point that defaulting to '4, EOS' could lead to confusion though 🤔